### ISSUE TITLE
Dirty floats don't work with non-string objects

### ISSUE THREAD

**Issue by amueller** [dabl/dabl#154]
If a dirty float column has floats encoded as strings and weird non-string objects, it works fine. But if it has floats and non-float objects the regex approach doesn't work and we don't recognize it as a dirty float column.

That's annoying!

e.g.
```python
[1, ('a', 'b'), 3, 4, 4, 4]
```
doesn't work, but
```python
['1', ('a', 'b'), '3', '4', '4', '4']
```

**Comment by amueller** [comment]
We could convert dtype object columns into strings at the very beginning and that might be the easiest way to deal with this?

**Comment by amueller** [comment]
Fixed in #162
